api/krusty/inlinelabels_test.go

```diff
@@ -70,6 +70,7 @@ spec:
     metadata:
       labels:
         a: b
+        c: d
```

Comment:
why does `foo: bar` not show up here?

Reply:
In this test, [`/app/deployment.yaml` is](https://github.com/kubernetes-sigs/kustomize/blob/923c21fe356dcbf1a1669e0a0629675a602c8539/api/krusty/inlinelabels_test.go#L12):
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-deployment
spec:
  template:
    spec:
      containers:
      - name: my-deployment
        livenessProbe:
          httpGet:
            path: /healthz
            port: 8080
---
apiVersion: example.dev/v1
kind: MyCRD
metadata:
  name: crd
```
^here the pod template has no metadata section. But apparently when kustomize runs and adds the `a: b` label, it creates the pod template metadata section. (`a: b` pair is configured with `includeSelectors: true`)
```
- pairs:
    a: b
  includeSelectors: true
```
After that, the rest of the labels are applied in metadata and template/metadata as expected. [The pair `foo: bar` is declared before `a: b`](https://github.com/kubernetes-sigs/kustomize/blob/923c21fe356dcbf1a1669e0a0629675a602c8539/api/krusty/inlinelabels_test.go#L41) therefore it never gets added to template/metadata

Reply:
@natasha41575 I fixed the test [by completing the initial deployment.yaml](https://github.com/kubernetes-sigs/kustomize/pull/4209/commits/a8a0e8c1553dbd64bf22b8d2de883634313ab597), as I believe `selector` and `template/metadata` are both required fields of the Deployment kind. Let me know if this is correct